File: src/main/java/com/github/unldenis/hologram/line/animated/TextALine.java
package com.github.unldenis.hologram.line.animated;

import com.github.unldenis.hologram.animation.Animation;

import java.util.Collection;
import java.util.Optional;
import org.bukkit.Location;
import org.bukkit.entity.Player;
import org.bukkit.plugin.Plugin;

public final class TextALine implements ITextLine, IAnimated {

  private final TextLine textLine;
  private final IAnimated animatedLine;

  public TextALine(TextLine textLine, IAnimated animatedLine) {
    this.textLine = textLine;
    this.animatedLine = animatedLine;
  }

  @Override
  public boolean isClickable() {
    return textLine.isClickable();
  }

  @Override
  public String parse(Player player) {
    return textLine.parse(player);
  }

  @Override
  public TextLine asTextLine() {
    return textLine;
  }

  @Override
  public PlaceholdersJava getPlaceholders() {
    return textLine.getPlaceholders();
  }

  @Override
  public Plugin getPlugin() {
    return textLine.getPlugin();
  }

  @Override
  public Type getType() {
    return Type.TEXT_ANIMATED_LINE;
  }

  @Override
  public int getEntityId() {
    return textLine.getEntityId();
  }

  @Override
  public Location getLocation() {
    return textLine.getLocation();
  }

  @Override
  public void setLocation(Location location) {
    textLine.setLocation(location);
  }

  @Override
  public String getObj() {
    return textLine.getObj();
  }

  @Override
  public void setObj(String obj) {
    textLine.setObj(obj);
  }

  @Override
  public void hide(Player player) {
    textLine.hide(player);
  }

  @Override
  public void teleport(Player player) {
    textLine.teleport(player);
  }

  @Override
  public void show(Player player) {
    textLine.show(player);
  }

  @Override
  public void update(Player player) {
    textLine.update(player);
  }

  @Override
  public void setAnimation(Animation animation, Collection<Player> seeingPlayers) {
    animatedLine.setAnimation(animation, seeingPlayers);
  }

  @Override
  public void removeAnimation() {
    animatedLine.removeAnimation();
  }

  @Override
  public Optional<Animation> getAnimation() {
    return animatedLine.getAnimation();
  }
}

File: src/main/kotlin/com/github/unldenis/hologram/line/animated/TextALine.kt
package com.github.unldenis.hologram.line.animated

import com.github.unldenis.hologram.animation.Animation
import com.github.unldenis.hologram.line.ILine
import com.github.unldenis.hologram.line.ITextLine
import com.github.unldenis.hologram.line.TextLine
import com.github.unldenis.hologram.placeholder.Placeholders
import org.bukkit.Location
import org.bukkit.entity.Player
import org.bukkit.plugin.Plugin
import java.util.*

class TextALine(override val textLine: TextLine, private val animatedLine: IAnimated) : ITextLine, IAnimated {
    override val clickable: Boolean
        get() = textLine.clickable
    override val placeholders: Placeholders
        get() = textLine.placeholders

    override fun parse(player: Player): String {
        return textLine.parse(player)
    }

    override val plugin: Plugin
        get() = textLine.plugin

    override val type: ILine.Type
        get() = ILine.Type.TEXT_ANIMATED_LINE

    override val entityId: Int
        get() = textLine.entityId

    override val location: Location?
        get() = textLine.location

    override var obj: String
        get() = textLine.obj
        set(value) {
            textLine.obj = value
        }

    override fun setLocation(value: Location) {
        textLine.setLocation(value)
    }

    override fun hide(player: Player) {
        textLine.hide(player)
    }

    override fun teleport(player: Player) {
        textLine.teleport(player)
    }

    override fun show(player: Player) {
        textLine.show(player)
    }

    override fun update(player: Player) {
        textLine.update(player)
    }

    override fun setAnimation(animation: Animation, seeingPlayers: Collection<Player>) {
        animatedLine.setAnimation(animation, seeingPlayers)
    }

    override fun removeAnimation() {
        animatedLine.removeAnimation()
    }

    override val animation: Optional<Animation>
        get() = animatedLine.animation
}
